scim/resource_user.go

```diff
@@ -93,7 +100,18 @@ func ResourceUser() *schema.Resource {
 			return NewUsersAPI(ctx, c).Update(d.Id(), u)
 		},
 		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			return NewUsersAPI(ctx, c).Delete(d.Id())
+			usersAPI := NewUsersAPI(ctx, c)
+
+			if d.Get("delete_repos").(bool) {
+				d.Set("cheese", "cheese")
+				_ = usersAPI.DeleteRepos(d.Id())
```

Comment:
1. scim can be account-level and workspace-level. how do you plan to check for that here?
2. do not swallow the error on deletion and if these properties are set, return error back to user. please be very careful for misinterpretation of non-found resources on workspace/repos to delete user from terraform state, but not deleting it from actual workspace.

Reply:
1. I don't see anything relating to workspace/account anywhere in the users go/resource/testing files. Would love a hint.
2. Noted very interesting point. Makes sense. My current delete function has 3 calls that could produce errors (image). Should I just return the first not `nil` error, else return `NewUsersAPI(ctx, c).Delete(d.Id())`? That seems like bad design but I'm not sure how to handle 3 potential errors. 
<img width="550" alt="image" src="https://user-images.githubusercontent.com/18635689/208607123-37563faf-4f49-4807-a642-e599afa3fbc1.png">

Reply:
Note to self. Once each error is accessible via the return, it will be easy to implement the below test.

Reply:
Wrap original error into another error.

Reply:
Chatted during office hours and we'll just return error if not nil - no wrapping.